drivers/sensor/lis2mdl/lis2mdl.c

```diff
@@ -148,7 +108,7 @@ static void lis2mdl_channel_get_temp(struct device *dev,
 	struct lis2mdl_data *drv_data = dev->driver_data;
 
 	val->val1 = drv_data->temp_sample / 100;
-	val->val2 = drv_data->temp_sample % 100;
+	val->val2 = (drv_data->temp_sample % 100) * 10000;
```

Comment:
Is this change really related to stmemsc i/f?

Reply:
No, I realized this was a bug. I can put it in a separate commit